### ISSUE TITLE
within_frame doesn't work if the iframe's name is set after the iframe was added to the document

### ISSUE THREAD

**Issue by tjgrathwell** [teampoltergeist/poltergeist#559]
Try out this test in your favorite app:

```ruby
require 'rails_helper'

describe "#within_frame", js: true do
  let (:iframe_url) { "http://echo.jsontest.com/sandwich/sunday" }
  let (:iframe_content) { "sandwich" }

  it "works if the iframe has a 'name' initially" do
    page.execute_script(<<-EOT)
      frame1 = document.createElement('iframe');
      frame1.name = 'some_frame';
      frame1.src = '#{iframe_url}';
      document.body.appendChild(frame1);
    EOT

    expect(page).to have_css('iframe[name=some_frame]')
    within_frame 'some_frame' do
      expect(page).to have_content(iframe_content)
    end
  end

  it "does not work if the iframe has a 'name' after being appended" do
    page.execute_script(<<-EOT)
      frame2 = document.createElement('iframe');
      frame2.src = '#{iframe_url}';
      document.body.appendChild(frame2);
      frame2.name = 'another_frame';
    EOT

    expect(page).to have_css('iframe[name=another_frame]')
    within_frame 'another_frame' do
      expect(page).to have_content(iframe_content)
    end
  end
end
```

Result:
```
1) #within_frame does not work if the iframe has a 'name' after being appended
   Failure/Error: within_frame 'another_frame' do
   Capybara::Poltergeist::FrameNotFound:
     The frame 'another_frame' was not found.
```

The only difference between the two tests is the order of document.body.append vs ```name = ```.

You can see that in both cases the have_css said a frame of that name was on the page, but it turned out it wasn't in a good enough state to do within_frame with. Fails in 1.5.1 and master, works in both capybara-webkit and selenium-webdriver.

**Comment by AMHOL** [comment]
Just came across this error today

**Comment by tmertens** [comment]
I am seeing this as well, it is a major blocker to using poltergeist.  `find('#some_iframe')` works when querying the page via Capybara; however, when I try to use `within_frame` and pass name, id, or the element found via `find`, Poltergeist always returns a `FrameNotFound` error.

**Comment by ChrisBAshton** [comment]
+1. Really need Poltergeist to be able to cope with dynamically added iframes!

If it could be done as part of the same refactor, I'd also really like to be able to use `within_frame the_iframe`, where `the_iframe` is a selector, or a frame name, or an Element.

**Comment by twalpole** [comment]
Should be fixed by PR #674 that was merged
